website/source/docs/secrets/mongodb/index.html.md

```diff
@@ -0,0 +1,444 @@
+---
+
+# MongoDB Secret Backend
+
+Name: `mongodb`
+
+The `mongodb` secret backend for Vault generates MongoDB database credentials
+dynamically based on configured roles. This means that services that need
+to access a MongoDB database no longer need to hard-code credentials: they
+can request them from Vault and use Vault's leasing mechanism to more easily
+roll them.
+
+Additionally, it introduces a new ability: with every service accessing
+the database with unique credentials, it makes auditing much easier when
+questionable data access is discovered: you can track it down to the specific
+instance of a service based on the MongoDB username.
+
+Vault makes use of its own internal revocation system to ensure that users
+become invalid within a reasonable time of the lease expiring.
+
+This page will show a quick start for this backend. For detailed documentation
+on every path, use `vault path-help` after mounting the backend.
+
+## Quick Start
+
+The first step to using the mongodb backend is to mount it.
+Unlike the `generic` backend, the `mongodb` backend is not mounted by default.
+
+```
+$ vault mount mongodb
+Successfully mounted 'mongodb' at 'mongodb'!
+```
+
+Next, we must tell Vault how to connect to MongoDB. This is done by providing
+a standard connection string (URI):
+
+```
+$ vault write mongodb/config/connection uri="mongodb://admin:Password!@mongodb.acme.com:27017/admin?ssl=true"
+Success! Data written to: mongodb/config/connection
+```
+
+In this case, we've configured Vault with the username `admin` and password
+`Password!`, connecting to an instance at `mongodb.acme.com` on port `27017`
+with TLS. The user must have privileges to manage users and their roles in the
+databases Vault will manage users in. The built-in role `userAdminAnyDatabase`
+is the simplest way to grant the necessary permissions if we want Vault to
+manage all users in all databases.
+
+Optionally, we can configure the lease settings for the credentials generated
+by Vault. This is done by writing to the `config/lease` key:
+
+```
+$ vault write mongodb/config/lease ttl=1h max_ttl=24h
+Success! Data written to: mongodb/config/lease
+```
+
+This restricts each user to being valid or leased for 1 hour at a time, with
+a maximum total use period of 24 hours. This forces an application to renew
+its credentials at least hourly and to recycle them once per day.
+
+The next step is to configure a role. A role is a logical name that maps
+to a policy used to generate MongoDB credentials for that role.
+
+Note that MongoDB also uses roles. The roles you define in Vault are distinct
+from the built-in and user-defined roles in MongoDB. In fact, when defining
+a Vault role you may specify the MongoDB roles that should be assigned to
+users created for that Vault role.
+
+For example, let's create a "readonly" role:
+
+```
+$ vault write mongodb/roles/readonly db=foo roles="[ \"readWrite\", { \"role\": \"read\", \"db\": \"bar\" } ]"
```

Comment:
I believe it is valid with Vault CLI key/value pairs to use single quotes, in which case you can simplify the `roles` parameter here since you won't have to escape the double quotes.